tests/unit/test_client.py

```diff
@@ -111,6 +112,16 @@ def test_client_signature_override(self, request_signer):
         request_signer.assert_called_with(
             mock.ANY, mock.ANY, mock.ANY, 'foo', mock.ANY, mock.ANY)
 
+    @mock.patch('botocore.client.RequestSigner')
```

Comment:
So there is another test that covers some of this:

https://github.com/boto/botocore/blob/develop/tests/unit/test_signers.py#L117-L127

I've also added a new test to ensure the request is attempted without any auth headers/query args.